### ISSUE TITLE
How to use pickneglogsoftmax() and softmax() for multitasking classifier by GPU?

### ISSUE THREAD

**Issue by yang182** [clab/dynet#1578]
I have a multitasking classifier problem, eg: dynet.LSTM output.value = [0.1, 0,2, 0,8, 0.5, 0.8, 0.2, 0.9, 0.1], the one to three is classifier task one, four to six is classifier task two and remain is classifier task three. How to use pickneglogsoftmax() and softmax() by GPU and cpu?  Or some other alternative method.

**Comment by neubig** [comment]
vals1 = dy.select_rows(x, [0, 1, 2])
    loss1 = dy.pickneglogsoftmax(vals1, label1)
    vals2 = dy.select_rows(x, [3, 4])
    loss2 = dy.pickneglogsoftmax(vals2, label2)
    vals3 = dy.select_rows(x, [5, 6])
    loss3 = dy.pickneglogsoftmax(vals3, label3)
    loss = loss1 + loss2 + loss3

would do it in this case.

**Comment by yang182** [comment]
Thanks  @neubig
